rollout/sync.go

```diff
@@ -656,7 +656,15 @@ func (c *rolloutContext) calculateRolloutConditions(newStatus v1alpha1.RolloutSt
 			if c.newRS != nil {
 				msg = fmt.Sprintf(conditions.ReplicaSetTimeOutMessage, c.newRS.Name)
 			}
-			condition := conditions.NewRolloutCondition(v1alpha1.RolloutProgressing, corev1.ConditionFalse, conditions.TimedOutReason, msg)
+
+			var condition *v1alpha1.RolloutCondition
+			// When ProgressDeadlineAbort is set, abort the update
+			if c.rollout.Spec.ProgressDeadlineAbort {
+				c.pauseContext.AddAbort(msg)
+				condition = conditions.NewRolloutCondition(v1alpha1.RolloutPaused, corev1.ConditionTrue, conditions.TimedOutReason, msg)
+			} else {
+				condition = conditions.NewRolloutCondition(v1alpha1.RolloutProgressing, corev1.ConditionFalse, conditions.TimedOutReason, msg)
```

Comment:
1. I don't think it's correct to conditionally add the Progressing=False. If we get to line 652 case block, we should **_always_** set Progressing to be false.

2. It doesnt look like we will emit an event about the abort, like we do in line 607. Can we structure the code such that if we abort here, we also emit the K8s event about the abort?

3. The setting of the condition `condition = conditions.NewRolloutCondition(v1alpha1.RolloutPaused, corev1.ConditionTrue, conditions.TimedOutReason, msg)` seems incorrect. Why would we be considered paused?

Reply:
Thanks. I was mistaken about the Paused and Progressing state. Please checkout the updated PR.